subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -240,7 +240,7 @@ private String internedToString(Object target) {
         }
 
         private boolean isAllowed(Path path, String name, boolean isDirectory, @Nullable BasicFileAttributes attrs, Iterable<String> relativePath) {
-            if (attrs == null) {
+            if (attrs == null || attrs.isOther()) {
```

Comment:
I'm wondering if it would be useful to add some comments here about our current thinking. The API here (wtf is `other`) is not very clear, and we might find some issues with this code later where understanding how we thought things should work originally might come in handy.

Reply:
Can you please add some Javadoc to this method please?

Reply:
this change doesn't exist anymore

Reply:
Would you still like me to add javadoc to this unchanged private method?

Reply:
That would be great, yes. You've dug into this deeper than anyone else did recently, it would be nice to surface some of what you found.

Reply:
Done